core/src/modules/configLoaders/index.ts

```diff
@@ -106,71 +109,75 @@ export async function processConfigObjects(
   };
   const errors: string[] = [];
 
+  configObjects = sortConfigurationObjects(configObjects);
+
   for (const i in configObjects) {
     const configObject = configObjects[i];
     if (Object.keys(configObject).length === 0) continue;
-    let klass = configObject?.class?.toLocaleLowerCase();
-    let object;
+    let klass = configObject?.class?.toLowerCase();
+    let guids: GuidsByClass;
     try {
       switch (klass) {
         case "setting":
-          object = await loadSetting(
-            configObject,
-            externallyValidate,
-            validate
-          );
+          guids = await loadSetting(configObject, externallyValidate, validate);
           break;
         case "app":
-          object = await loadApp(configObject, externallyValidate, validate);
+          guids = await loadApp(configObject, externallyValidate, validate);
           break;
         case "source":
-          object = await loadSource(configObject, externallyValidate, validate);
-          if (configObject.bootstrappedProperty) {
-            seenGuids["property"].push(
-              await validateAndFormatGuid(
-                Property,
-                configObject.bootstrappedProperty.id
-              )
-            );
-          }
+          guids = await loadSource(configObject, externallyValidate, validate);
           break;
         case "property":
-          object = await loadProperty(
+          guids = await loadProperty(
             configObject,
             externallyValidate,
             validate
           );
           break;
         case "group":
-          object = await loadGroup(configObject, externallyValidate, validate);
+          guids = await loadGroup(configObject, externallyValidate, validate);
           break;
         case "schedule":
-          object = await loadSchedule(
+          guids = await loadSchedule(
             configObject,
             externallyValidate,
             validate
           );
           break;
         case "destination":
-          object = await loadDestination(
+          guids = await loadDestination(
             configObject,
             externallyValidate,
             validate
           );
           break;
         case "apikey":
-          object = await loadApiKey(configObject, externallyValidate, validate);
+          guids = await loadApiKey(configObject, externallyValidate, validate);
           break;
         case "team":
-          object = await loadTeam(configObject, externallyValidate, validate);
+          guids = await loadTeam(configObject, externallyValidate, validate);
           break;
         case "teammember":
-          object = await loadTeamMember(
+          guids = await loadTeamMember(
             configObject,
             externallyValidate,
             validate
           );
           break;
+        case "synctable":
+          const many = await expandSyncTable(
```

Comment:
ideally, I want to `expand` first later somehow (to get "base" configs), but there is a dependecy issue because it actually needs the app. so more dynamic DAG work is needed.